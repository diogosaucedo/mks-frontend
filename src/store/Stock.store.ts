import { PayloadAction, createSlice } from '@reduxjs/toolkit';

import { ICardProduct } from '@/interfaces';

const stock = createSlice({
  name: 'stock',
  initialState: {
    counter: 0,
    total: 0,
    visible: false,
    products: [] as never | ICardProduct[],
  },
  reducers: {
    toggleVisible(state) {
      state.visible = !state.visible;
    },
    addProduct(state, action: PayloadAction<ICardProduct>) {
      if (
        state.products.find(
          (item: ICardProduct) => item.id == action.payload.id,
        )
      ) {
      } else {
        state.products.push(action.payload);
      }
    },
    removeProduct(state, action: PayloadAction<number>) {
      state.products = state.products.filter(
        (product) => product.id !== action.payload,
      );
    },
    counter(state) {
      state.counter = state.products.length;
    },
    getTotal(state) {
      state.total = state.products.reduce(
        (acc, item) => acc + item.amount * item.price,
        0,
      );
    },
    incrementAmountProduct(state, action: PayloadAction<number>) {
      state.products = state.products.map((product) => {
        if (product.id === action.payload) {
          return { ...product, amount: product.amount + 1 };
        } else {
          return product;
        }
      });
    },
    decrementAmountProduct(state, action: PayloadAction<number>) {
      state.products = state.products.map((product) => {
        if (product.id === action.payload && product.amount > 1) {
          return { ...product, amount: product.amount - 1 };
        } else {
          return product;
        }
      });
    },
  },
});

export const {
  toggleVisible,
  addProduct,
  counter,
  removeProduct,
  getTotal,
  incrementAmountProduct,
  decrementAmountProduct,
} = stock.actions;
export default stock.reducer;
