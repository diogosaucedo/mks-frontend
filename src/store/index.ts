import { configureStore } from '@reduxjs/toolkit'
import stockStore from './Stock.store'

export const store = configureStore({
  reducer: {
    stock: stockStore
  },
})

// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch
