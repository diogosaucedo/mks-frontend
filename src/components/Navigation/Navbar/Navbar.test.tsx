import { render, screen } from '@testing-library/react';
import React from 'react';
import { NavBar } from '.';
import { store } from '@/store/';
import { Provider } from 'react-redux';

describe('Navbar component', () => {
  test('should render properties correctly', () => {
    render(
      <Provider store={store}>
        <NavBar />
      </Provider>,
    );

    expect(screen.getByText('MKS')).toBeInTheDocument();
    expect(screen.getByText('Sistemas')).toBeInTheDocument();
    expect(screen.getByRole('button')).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <NavBar />
      </Provider>,
    );

    expect(asFragment().firstChild).toMatchSnapshot();
  });
});
