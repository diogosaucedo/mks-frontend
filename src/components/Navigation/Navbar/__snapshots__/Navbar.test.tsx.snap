// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component should match snapshot 1`] = `
<nav
  class="sc-jSUZER gkjDCd"
  height="100px"
  width="100%"
>
  <div
    class="sc-ftTHYK intCNY"
    width="90%"
  >
    <div
      class="sc-hLBbgP hgsTQh"
    >
      <h1
        class="sc-bcXHqe dFqMNp"
        color="#FFFFFF"
        font-size="40px"
        font-weight="600"
        type="h1"
      >
        MKS
      </h1>
      <p
        class="sc-gswNZR hRerhq"
        color="#FFFFFF"
        font-size="20px"
        font-weight="300"
      >
        Sistemas
      </p>
    </div>
    <button
      class="sc-dkrFOg gulMTY"
      height="45px"
      width="90px"
    >
      <div
        class="sc-hLBbgP fEIcqa"
      >
        <img
          alt="Icone do carrinho de compras"
          data-nimg="1"
          decoding="async"
          height="18"
          loading="lazy"
          src="./cart.svg"
          style="color: transparent;"
          width="18"
        />
        <p
          class="sc-gswNZR iFEBoy"
          font-size="18px"
          font-weight="700"
        >
          0
        </p>
      </div>
    </button>
  </div>
</nav>
`;
